pkg/events/sorting/pool.go

```diff
@@ -0,0 +1,78 @@
+package sorting
+
+import (
+	"fmt"
+	"github.com/aquasecurity/tracee/pkg/external"
+	"sync"
+)
+
+const poolFreeingPart = 2
+
+// A struct used to store empty nodes to prevent repeated allocation and freeing from the heap
+type eventsPool struct {
+	head             *eventNode
+	poolMutex        sync.Mutex
+	allocationsCount int
+	poolSize         int
+}
+
+func (p *eventsPool) Alloc(event *external.Event) *eventNode {
+	p.poolMutex.Lock()
+	p.allocationsCount += 1
+	defer p.poolMutex.Unlock()
+	node := p.head
+	if node != nil && node.isAllocated == false {
+		p.head = node.previous
+		node.event = event
+		node.isAllocated = true
+		node.previous = nil
+		p.poolSize -= 1
+		return node
+	}
+	return &eventNode{event: event, isAllocated: true}
+}
+
+func (p *eventsPool) Free(node *eventNode) error {
+	p.poolMutex.Lock()
+	defer p.poolMutex.Unlock()
+	// Prevent malicious use of free
+	if p.allocationsCount == 0 {
+		return fmt.Errorf("a free operation on node from pool is requested when there are supposingly no allocated nodes")
+	}
+
+	node.isAllocated = false
+	node.previous = nil
+	node.next = nil
+	p.allocationsCount -= 1
+	// Free memory in case of pooling too many nodes
+	if p.poolSize >= p.allocationsCount {
+		freeingAmount := p.poolSize / poolFreeingPart
```

Comment:
Should we set a minimum here ? I'm afraid of being stuck in lower values depending on alloc/dealloc sequence behavior. Something like if `p.poolSize > 200` && >= `p.allocationsCount`. WDYT ?